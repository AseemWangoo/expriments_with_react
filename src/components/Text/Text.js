import React from "react";
import PropTypes from 'prop-types'

const Text = ({ text }) => {
  return (
    <div data-testid="text">
      <h3>{text}</h3>
    </div>
  );
};

Text.defaultProps = {
  text: "Hey world",
};

Text.propTypes = {
  text: PropTypes.string,
};

export default Text;
