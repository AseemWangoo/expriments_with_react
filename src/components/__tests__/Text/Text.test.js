import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Text from "../../Text/Text";
import "@testing-library/jest-dom";

afterEach(() => {
  cleanup();
});

test("should render text component", () => {
  render(<Text />);

  var textElem = screen.getByTestId("text");

  expect(textElem).toBeInTheDocument();
});

test("text matches snapshot", () => {
  const component = renderer.create(<Text />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});