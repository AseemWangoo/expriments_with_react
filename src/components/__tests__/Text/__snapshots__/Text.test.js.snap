// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text matches snapshot 1`] = `
<div
  data-testid="text"
>
  <h3>
    Hey world
  </h3>
</div>
`;
